### PR TITLE
refactor(types): Improve types coverage & docs for js parser

### PR DESCRIPTION
**Pull request by TheLarkInn**
## Summary
Increase type coverage for the operand handlers and helper functions of UnaryExpression and BinaryExpression hooks in the JavaScript parser. 

### <samp>🤖 Generated by Copilot at 0fff38d</samp>

This pull request enhances the documentation and readability of the `BasicEvaluatedExpression` and `JavascriptParser` modules, which are responsible for parsing and evaluating JavaScript code in webpack. It adds JSDoc comments, template annotations, and parameter name changes to various functions and methods, and simplifies some logic and evaluation cases.

### <samp>🤖 Generated by Copilot at 0fff38d</samp>

*  Add JSDoc comments to various functions and methods in `lib/javascript/JavascriptParser.js` and `lib/javascript/BasicEvaluatedExpression.js` to improve documentation and readability ([link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-5059a8f5a078630e6d1a2f751349f83f201e63c3c4ae9d2b9d1d4f274696c6cdR225-R228), [link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-5059a8f5a078630e6d1a2f751349f83f201e63c3c4ae9d2b9d1d4f274696c6cdR254-R257), [link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2R105-R122), [link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2R129-R144), [link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2R153-R162), [link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2R516-R558), [link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2R614-R621), [link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2R665-R669), [link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2R723-R727))
*  Add template annotations and rename parameters to the `handleConstOperation` function in `lib/javascript/JavascriptParser.js` to improve type checking and generality ([link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2L502-R596), [link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2L509-R603), [link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2L828-R950))
*  Update the call to the `handleConstOperation` function in `lib/javascript/JavascriptParser.js` to reflect the parameter name change ([link](https://github.com/webpack/webpack/pull/17094/files?diff=unified&w=0#diff-561daa0feb66d49c350f9f8296e8f9bfe50a7ed5efbb4226d2b512aa20d201f2L509-R603))

### CODE REVIEW

**Comment by TheLarkInn** [comment]
@chenxsan Despite believing that most of the documentation I've provided here is contributor specific, I going to have the bot create an issue for documentation because I believe that most people don't know that webpack's parser will evaluate simple expressions for many module / import specific scenarios (as I'm remembering/learning throughout this process).  

If you personally don't find any value in actually documenting this, then you can close the issue.

**Review by alexander-akait — COMMENTED** [review]
Looks update our types and we can merge :star:

**Comment by TheLarkInn** [comment]
Very small inclusion of types in Basic Evaluated Expression. I will include better documentation for this in the future.

File: types.d.ts [hunk 1]
```diff
@@ -525,7 +525,15 @@ declare abstract class BasicEvaluatedExpression {
 	 * Can this expression have side effects?
 	 */
 	couldHaveSideEffects(): boolean;
-	asBool(): any;
```
Comment (alexander-akait) [inline types.d.ts:528]: Less `any` :star:

**Comment by chenxsan** [comment]
> @chenxsan Despite believing that most of the documentation I've provided here is contributor specific, I going to have the bot create an issue for documentation because I believe that most people don't know that webpack's parser will evaluate simple expressions for many module / import specific scenarios (as I'm remembering/learning throughout this process).
> 
> If you personally don't find any value in actually documenting this, then you can close the issue.

Yup, it's very helpful to contributors or who want to dive into the source code, regular users might not notice it. I'll skip it for this time, we can always come back to document them if there're requests from regular users.

**Pull request merged**